client/signup/steps/design-type-with-store/bluehost-store/index.jsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import identity from 'lodash/identity';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+import StepHeader from 'signup/step-header';
+import Button from 'components/button';
+import { abtest } from 'lib/abtest';
+import { localize } from 'i18n-calypso';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+function getPrice() {
+	return 'bluehost' === abtest( 'signupStoreBenchmarking' ) ? '$3.95' : '$12.95';
+}
+
+export const BluehostStoreStep = props => {
+	const {
+		onBackClick,
+		translate,
+		recordPartnerClick,
+		partnerName,
+		partnerUrl
+	} = props;
+
+	return (
+		<div>
+			<StepHeader
+				headerText={ translate( 'Create a WordPress Store' ) }
+				subHeaderText={ translate( 'Our partners at BlueHost and WooCommerce are here for you' ) }
+			/>
+
+			<div className="design-type-with-store__container">
+				<div className="design-type-with-store__copy">
+					<img src="/calypso/images/signup/bluehost-logo.png" className="design-type-with-store__bluehost-logo" />
+					{ translate(
+						'We\'ve partnered with BlueHost, a top-notch WordPress hosts ' +
+						'with a knack for building great e-commerce stores using WooCommerce.'
+					) }
+				</div>
+
+				<div className="design-type-with-store__form">
+					<span className="design-type-with-store__price-text"> { translate( 'Starting at' ) } </span>
+					<span className="design-type-with-store__price"> <b>{ getPrice() }</b>/mo </span>
+
+					<Button
+						primary
+						className="design-type-with-store__form-submit"
+						target="_blank"
+						href={ partnerUrl }
+						onClick={ () => recordPartnerClick( partnerName ) }
```

Comment:
As this will throw an error for our linting (, there are a few ways around it:

One way is to create this closure in `render()` above the `return`. This seems a bit silly to me.

The better way here is to probably create it in the `connect()` function <del>so that it's not being recreated on every call to `render()`</del> <ins>Actually it gets recreated regardless because we're passing (necessarily) the `partnerName` each time</ins>. The only drawback here is that the naming could be misleading; that can be improved if necessary

```js
const mapDispatchToProps = dispatch => ( {
	partnerClickRecorder: partnerName => () => dispatch( … )
} );
```

This adds one bit of indirection because we end up doing `() => () => {}` but it's not too bad.